curriculum/challenges/english/07-scientific-computing-with-python/learn-special-methods-by-building-a-vector-space/65f9b9710cca621244d3bde1.md

```diff
@@ -0,0 +1,58 @@
+---
+id: 65f9b9710cca621244d3bde1
+title: Step 40
+challengeType: 20
+dashedName: step-40
+---
+
+# --description--
+
+In Python, `NotImplemented` is a special value used to indicate that an operation is not implemented for a specific case.
+
+`NotImplemented` does not raise an exception immediately. Instead, it signals to ask to the other operand how to perform the operation. If the request cannot be satisfied, a `TypeError` is returned by default.
+
+Because you want to be able to sum two vectors only if they have the belong to the same class, return `NotImplemented` from the `if` statement you created in the previous step.
```

Comment:
```suggestion
Because you want to be able to sum two vectors only if they belong to the same class, return `NotImplemented` from the `if` statement you created in the previous step.
```